substrate/primitives/metadata-ir/src/types.rs

```diff
@@ -340,19 +343,29 @@ impl IntoPortable for StorageEntryTypeIR {
 pub struct PalletCallMetadataIR<T: Form = MetaForm> {
 	/// The corresponding enum type for the pallet call.
 	pub ty: T::Type,
+	/// Deprecation status of the pallet call itself
+	pub deprecation_info: DeprecationStatus<T>,
+	/// Deprecation status of the call indexes
+	pub deprecated_indexes: BTreeMap<usize, DeprecationStatus<T>>,
 }
```

Comment:
Could we group the deprecation infos? Some initial idea would be:

```rust
enum DeprecationMetadataIr {
    Fully(DeprecationStatusIr),
    Partially(BTreeMap<usize, DeprecationStatusIr<T>>),
}
```

I would also rename the `DeprecationStatus` to `DeprecationStatusIR` sometime, since we'll need to convert this to a frame_metadata::DeprecationStatus

Reply:
Can i do this after this MR, to apply the change to every other place of usage and updating the tests?

Reply:
Yep sounds like a plan!